### PR TITLE
Fix TbSubscriptionUtils order for matching KeyValueType and DataType

### PR DESCRIPTION
**Pull request by AndriiLandiak**
## Pull Request description

Put your PR description here instead of this sentence.   

## General checklist

- [ ] You have reviewed the guidelines [document](https://docs.google.com/document/d/1wqcOafLx5hth8SAg4dqV_LV3un3m5WYR8RdTJ4MbbUM/edit?usp=sharing).
- [ ] [Labels](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/managing-labels#about-labels) that classify your pull request have been added.
- [ ] The [milestone](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/about-milestones) is specified and corresponds to fix version.  
- [ ] Description references specific [issue](https://github.com/thingsboard/thingsboard/issues).
- [ ] Description contains human-readable scope of changes.
- [ ] Description contains brief notes about what needs to be added to the documentation.
- [ ] No merge conflicts, commented blocks of code, code formatting issues.
- [ ] Changes are backward compatible or upgrade script is provided.
- [ ] Similar PR is opened for PE version to simplify merge. Crosslinks between PRs added. Required for internal contributors only.
  
## Front-End feature checklist

- [ ] Screenshots with affected component(s) are added. The best option is to provide 2 screens: before and after changes;
- [ ] If you change the widget or other API, ensure it is backward-compatible or upgrade script is present.
- [ ] Ensure new API is documented [here](https://github.com/thingsboard/thingsboard-ui-help)

## Back-End feature checklist

- [ ] Added corresponding unit and/or integration test(s). Provide written explanation in the PR description if you have failed to add tests.
- [ ] If new dependency was added: the dependency tree is checked for conflicts.
- [ ] If new service was added: the service is marked with corresponding @TbCoreComponent, @TbRuleEngineComponent, @TbTransportComponent, etc.
- [ ] If new REST API was added: the RestClient.java was updated, issue for [Python REST client](https://github.com/thingsboard/thingsboard-python-rest-client) is created.
- [ ] If new yml property was added: make sure a description is added (above or near the property).

### CODE REVIEW

File: application/src/test/java/org/thingsboard/server/utils/TbSubscriptionUtilsTest.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+import org.junit.Test;
+import org.thingsboard.server.common.data.kv.DataType;
+import org.thingsboard.server.service.subscription.TbSubscriptionUtils;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TbSubscriptionUtilsTest {
+
+    @Test
```
Comment (smatvienko-tb) [inline application/src/test/java/org/thingsboard/server/utils/TbSubscriptionUtilsTest.java:26]: Parametrized test, enum source

**Review by smatvienko-tb — COMMENTED** [review]
As far as I understand this is similar to the KeyValueType order problem
https://github.com/thingsboard/thingsboard/pull/9691

As far as I see, the KeyValueProto usage was replaced with a DataType. That makes sense to me. I would like to have a deprecated annotation on unused proto

Please, check whether or not we are using KeyValueType protos in Attribute caching. If yes. please, add attribute cache clearing procedure in the upgrade script.

File: application/src/test/java/org/thingsboard/server/utils/TbSubscriptionUtilsTest.java (hunk 1 above)
Comment (ViacheslavKlimov) [reply application/src/test/java/org/thingsboard/server/utils/TbSubscriptionUtilsTest.java:26]: for 3.7

**Comment by ashvayka** [comment]
This PR breaks backwards-compatibility and needs to be reverted.

**Comment by smatvienko-tb** [comment]
As we discussed in the latest meeting, these changes are only in use by subscription services, so it is a run-time and no breaking changes after all subscribers-submitters restart

**Comment by ViacheslavKlimov** [comment]
@ashvayka 
We will revert this for hotfix/3.6.3, due to possible issues during rolling update. The fix will wait for upgrade to 3.7.

**Pull request merged**